widgets-samples/cc/samples-cc-react-app/src/App.tsx

```diff
@@ -78,6 +81,54 @@ function App() {
     setSelectedWidgets((prev) => ({...prev, [name]: checked}));
   };
 
+  const changeAgentState = (newState: string) => {
+    // If the selected state is "Available", we pass it as is; otherwise, we consider it "Idle".
+    const chosenState = newState === 'Available' ? 'Available' : 'Idle';
+    // In the idle codes, we need to search for the 'Idle' state with code name 'Meeting'.
+    const lookupCodeName = newState === 'Available' ? 'Available' : 'Meeting';
+    
+    const idleCode = store.idleCodes?.find((code: any) => code.name === lookupCodeName);
+    if (!idleCode) {
+      console.error('No idle code found for selected state:', newState);
+      return;
+    }
+    const agentId = store.agentId || '';
+    store.cc
+      .setAgentState({
+        state: chosenState,
+        auxCodeId: idleCode.id,
+        agentId,
+        lastStateChangeReason: newState,
+      })
+      .then((response) => {
+        store.setCurrentState(response.data.auxCodeId);
+        store.setLastStateChangeTimestamp(new Date(response.data.lastStateChangeTimestamp));
+        console.log('Agent state updated to', chosenState);
+      })
+      .catch((error) => {
+        console.error('Error updating agent state:', error);
+      });
+  };
+
+  const handlePopoverSubmit = () => {
+    if (selectedState) {
+      changeAgentState(selectedState);
+    }
+    setShowRejectedPopup(false);
+    setSelectedState('');
+  };
+
+  useEffect(() => {
+    store.onTaskRejected = (reason: string) => {
+      setRejectedReason(reason);
+      setShowRejectedPopup(true);
+    };
```

Comment:
```suggestion
    const handleTaskRejected =(reason: string) => {
      setRejectedReason(reason);
      setShowRejectedPopup(true);
    };
    
    store.setOnTaskRejected(handleTaskRejected) 
```

Lets create a setter for the callBack rather than asking users to set a state in the store.